### PR TITLE
feature issue #3804

### PR DESCRIPTION
**Pull request by chuntaojun**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

#3804 

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: sys/src/main/java/com/alibaba/nacos/sys/Main.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package com.alibaba.nacos.sys;
+
+/**
+ * @author <a href="mailto:liaochuntao@live.com">liaochuntao</a>
+ */
+public class Main {
```
Comment (KomachiSion) [inline sys/src/main/java/com/alibaba/nacos/sys/Main.java:22]: What's the reason create this class?

File: sys/src/main/java/com/alibaba/nacos/sys/env/NacosDefaultPropertySourceEnvironmentPostProcessor.java [hunk 2]
```diff
@@ -1,20 +1,20 @@
 /*
- * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *  Copyright 1999-2018 Alibaba Group Holding Ltd.
```
Comment (KomachiSion) [inline sys/src/main/java/com/alibaba/nacos/sys/env/NacosDefaultPropertySourceEnvironmentPostProcessor.java]: Why change the indentation of lecense?

File: console/src/main/resources/application.properties [hunk 3]
```diff
@@ -175,3 +175,14 @@ nacos.istio.mcp.server.enabled=false
...
+
+## HTTP Encoding
+spring.http.encoding.force=true
+spring.http.encoding.enabled=true
+
+## i18n
+spring.messages.encoding=UTF-8
+
+## Exclude Spring Boot Auto-Configuration class(es)
+spring.autoconfigure.exclude=org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration,org.springframework.boot.autoconfigure.jdbc.DataSourceTransactionManagerAutoConfiguration
```
Comment (KomachiSion) [inline console/src/main/resources/application.properties:188]: Why exclude these classes？

File: sys/src/main/java/com/alibaba/nacos/sys/env/NacosAutoRefreshPropertySourceLoader.java [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    private Map<String, ?> loadProperties(Resource resource) throws IOException {
+        return new OriginTrackedPropertiesLoader(resource).load();
+    }
+    
+    @JustForTest
+    protected Map<String, Object> getProperties() {
+        return properties;
+    }
+    
+    static class OriginTrackedPropertiesLoader {
```
Comment (KomachiSion) [inline sys/src/main/java/com/alibaba/nacos/sys/env/NacosAutoRefreshPropertySourceLoader.java:105]: I found this inner class only used in current class line 97. And each time create a new one. Can this inner class refactor to some private method.

If there are other same package class need this class. Whether extract as an single class will be better?

File: sys/src/main/java/com/alibaba/nacos/sys/env/NacosDefaultPropertySourceEnvironmentPostProcessor.java [hunk 5]
```diff
@@ -56,7 +57,7 @@
      * @see ResourcePatternResolver#CLASSPATH_ALL_URL_PREFIX
      */
     public static final String RESOURCE_LOCATION_PATTERN =
-            CLASSPATH_ALL_URL_PREFIX + "META-INF/nacos-default.properties";
+            ResourcePatternResolver.CLASSPATH_ALL_URL_PREFIX + "META-INF/nacos-default.properties";
```
Comment (KomachiSion) [inline sys/src/main/java/com/alibaba/nacos/sys/env/NacosDefaultPropertySourceEnvironmentPostProcessor.java:60]: This file has been deleted?

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: sys/src/main/java/com/alibaba/nacos/sys/Main.java (hunk 1 above)
Comment (chuntaojun) [reply sys/src/main/java/com/alibaba/nacos/sys/Main.java:22]: already remove, wait next comit

File: sys/src/main/java/com/alibaba/nacos/sys/env/NacosDefaultPropertySourceEnvironmentPostProcessor.java (hunk 2 above)
Comment (chuntaojun) [reply sys/src/main/java/com/alibaba/nacos/sys/env/NacosDefaultPropertySourceEnvironmentPostProcessor.java]: already fix

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**